### ISSUE TITLE
Can nanopolish call variants with fast5s and reads from scrappie

### ISSUE THREAD

**Issue by mchaisso** [jts/nanopolish#299]
Hi, I have a "reference genome" (a p-read from a Canu assembly) that I would like to polish.  The reads are called using scrappie since I don't have access to Albacore.  The command line is straight forward:

When 
`nanopolish  variants  --reads=reads.fasta --bam=reads.bam --genome=longest.fasta  --consensus=longest.consensus.fasta --ploidy=2 --outfile=vars.vcf --models-fofn models.txt ` 
is executed, vars.vcf has no variants in it, and the consensus is identical to the genome.  There should probably be some differences because the "genome" has many errors.

I'm not sure if the albacore/scrappy difference is a red herring. 

Thanks,
-Mark

**Comment by jts** [comment]
Hi Mark,

It _should_ work with scrappie but I haven't tested it. The important thing is that the reads are named correctly - nanopolish expects the read name to be a guid which matches what is in the fast5 file.

A couple of comments on your command:

- -`--ploidy 2` shouldn't be used with `--consensus` as the consensus algorithm is not yet diploid-aware
- -`--models` is not necessary, that option is really only for debugging/development
- starting from a read sequence will be difficult, as nanopolish is designed to make small changes to the genome (ie going from 99% identity to 99.9%). Still, i would expect some changes to be made.

Could you paste the very last line printed to stderr? It contains QC information that may be helpful in figuring out what is going on.

Thanks,
Jared

**Comment by mchaisso** [comment]
Thanks for the comments. I removed `--ploidy 2 ` and `--models ...`, and added --verbose. The vcf file still just has the header and there is no verbose output.  I'm sure there is some massive user error here, my apologies in advance.

**Comment by jts** [comment]
Depending on the size of the region you are polishing it can take awhile before output shows up. Maybe try a small test window by adding `-w <contig name>:<start>-<end>`?

**Comment by mchaisso** [comment]
The contig is small and nanopolish runs almost instantly. I tried limiting
to 40 kbp with -w, but similarly no output.

-Mark

**Comment by jts** [comment]
Could you paste the post-run summary line? It will look like this:

```
[post-run summary] total reads: 2886, unparseable: 0, qc fail: 1, could not calibrate: 0, no alignment: 108, bad fast5: 0
```

**Comment by mchaisso** [comment]
So far no summary is listed, the program silently ends. The sample data is
small (~120M of all fast5s), I can set up a tar of the test case if that
helps (and PM it).

/home/cmb-16/mjc/shared/software_packages/nanopolish/nanopolish --version
nanopolish version 0.8.5
Written by Jared Simpson.

Copyright 2015-2017 Ontario Institute for Cancer Research

**Comment by jts** [comment]
Hmm, the summary should be printed. Happy to take a look at the reads if you PM me a link.

jared

**Comment by jts** [comment]
Thanks for the test data, I was able to track down the problem. Scrappie names the reads using the fast5 file name:

```
gridion00_20170502_FNFAE32997_MN00002_sequencing_throughput_GridION_ONTHG1_89225_ch98_read8805_strand.fast5
```

whereas albacore uses a guid like this:

```
004cf392-4019-4bf9-91ff-760df32b9f65
```

nanopolish expects the latter when it makes the `.readdb` file that links each basecalled read to its corresponding signal data. In your case it couldn't find the signal data for the scrappie reads so it silently ignored everything. I've just changed how the QC stats are calculated to make it clear that all reads are ignored when this happens.

If you want to continue with the scrappie reads you'll have to make your own `.readdb` file - this should be simple though as the scrappie read names are the fasf5 file names. Each entry should look like this:

```
gridion00_20170502_FNFAE32997_MN00002_sequencing_throughput_GridION_ONTHG1_89225_ch98_read8805_strand.fast5 fast5s/gridion00_20170502_FNFAE32997_MN00002_sequencing_throughput_GridION_ONTHG1_89225_ch98_read8805_strand.fast5
```

Jared
